### DISCUSSION TITLE
When I was using the ItemsPanelTemplate of ItemsControl, I found that I couldn't access the component in the backend code.

### DISCUSSION THREAD

**Discussion by tangxiaoqi-tangxiao** [AvaloniaUI/Avalonia#17597] [Q&A]
code：
        <ScrollViewer Grid.Row="0" VerticalScrollBarVisibility="Auto">
            <ItemsControl ItemsSource="{Binding FileItems}">
                <ItemsControl.ItemsPanel>
                    <ItemsPanelTemplate>
                        <UniformGrid x:Name="GridContentColumn" Rows="0" Columns="4" Margin="10" HorizontalAlignment="Stretch" VerticalAlignment="Top"/>
                    </ItemsPanelTemplate>
                </ItemsControl.ItemsPanel>
                <ItemsControl.ItemTemplate>
                    <DataTemplate>
  
                    </DataTemplate>
                </ItemsControl.ItemTemplate>
            </ItemsControl>
        </ScrollViewer>
        
        
back-end code：
private void OnSizeChanged(object? sender, SizeChangedEventArgs e)
    {
        var newWidth = e.NewSize.Width;
        if (GridContentColumn==null)
            return;
        if (newWidth <= ScreenSize.Md)
        {
            GridContentColumn.Columns = 3;
        }
        else if (newWidth <= ScreenSize.Lg)
        {
            GridContentColumn.Columns = 4;
        }
        else if (newWidth <= ScreenSize.Xl)
        {
            GridContentColumn.Columns = 6;
        }
        else
        {
            GridContentColumn.Columns = 8;
        }
    }
    
The GridContentColumn is always null.

I roughly understand that the components generated by ItemsControl do not include UniformGrid, so how should I obtain the UniformGrid component?

**Comment by msneijders** [comment]
Get it indirectly from the `<ItemsControl x:Name="items" />`, using `items.ItemsPanelRoot` -- that should be the instance of your `ItemsPanel`.
